src/app/Topology/Actions/NodeActions.tsx

```diff
@@ -139,11 +141,74 @@ export const nodeActions: NodeAction[] = [
   },
   { isSeparator: true },
   {
-    action: (element, { services }, track) => {
+    action: (element, { services }) => {
       const targetNode: TargetNode = element.getData();
-      track(services.api.deleteTarget(targetNode.target).subscribe(() => undefined));
+      services.api.deleteTarget(targetNode.target).subscribe(() => undefined);
     },
     title: 'Delete Target',
     includeList: [NodeType.CUSTOM_TARGET],
   },
+  {
+    action: (element, { services }) => {
+      const group: EnvironmentNode = element.getData();
+      const recordingName = `${group.name.replace(/[\s+-]/g, '_')}_${Date.now()}`;
+      services.api
+        .graphql(
+          `
+            query StartRecordingForGroup($filter: EnvironmentNodeFilterInput) {
+              environmentNodes(filter: $filter) {
+                  descendantTargets {
+                      doStartRecording(recording: {
+                          name: "${recordingName}",
+                          template: "Profiling",
+                          templateType: "TARGET",
+                          duration: 30
```

Comment:
In the absence of any UI widget for specifying a template and duration, I would lean more toward using the continuous template since the potential runtime impact (recording overhead) is lower, and probably with an unlimited duration or perhaps just something fairly long, like 10 or 15 minutes. I think if there were also an action for stopping recordings on the group then it would be fine to let the duration be unlimited.

Reply:
> I think if there were also an action for stopping recordings on the group then it would be fine to let the duration be unlimited.

Sure!! But about this, `Stop Recordings` here means stopping every recordings? Wondering how the users would know which recordings will be stopped? And how do we find the recording name?

Reply:
See other comment that I was writing concurrently

Reply:
> And how do we find the recording name?

I guess we can use well known name but that would allow creating once unless deleting the recordings later?

Reply:
Ahh nicee! Okayyy! I will change the query to this.

Reply:
So I think there would be:

- a Start Recording group action that uses a well-known name and sets the `restart` setting
- a Stop Recording group action that stops any recording with that name, ignoring any failures due to recording already being stopped
- a Delete/Remove Recording group action that deletes active recordings with that name regardless of running state

There could also hypothetically be a query that checks for the aggregate count of number of recordings with the well-known name that are currently in the Running state, and this can control whether the Stop Recording action is enabled. Might be overkill. Likewise for Remove Recording, checking the aggregate count of recordings with that name in any state.

Reply:
Oh, another idea, you can set that `${group.name.replace()}` and the timestamp and other details in the metadata when you perform the recording start mutation. The `metadata` should be provided as a (JSON) object with a `labels` field that is just a `Map<String, String>`. I think it goes like this:

```graphql
query {
    environmentNodes(filter: { name: "JDP" }) {
        descendantTargets {
            doStartRecording(recording: {
                name: "foo",
                template: "Profiling",
                templateType: "TARGET",
                duration: 30,
                metadata: {
                    labels: {
                        "cryostat.io/topology-group": "my-group-name"
                    }
                }
                }) {
                    name
                    state
            }
        }
    }
}
```

Reply:
Hey actually, I was trying to get this query to work but seems to throw invalid syntax or casting error

```graphql
query StartRecordingForGroup($filter: EnvironmentNodeFilterInput, $name: String!) {
              environmentNodes(filter: $filter) {
                  descendantTargets {
                      doStartRecording(recording: {
                          name: $name,
                          template: "Continuous",
                          templateType: "TARGET",
                          duration: 0,
                          restart: true,
                          metadata: {
                            labels: {
                                "cryostat.io/topology-group": "my-group-name"
                            }
                          },
                          }) {
                              name
                              state
                      }
                  }
              }
          }
```

Reply:
### With Labels
```
Invalid Syntax : offending token '\"cryostat.io/topology-group\"' at line 1 column 283
```

### With restart

```
Exception while fetching data (/environmentNodes[0]/descendantTargets[0]/doStartRecording) : class java.lang.Boolean cannot be cast to class java.lang.String (java.lang.Boolean and java.lang.String are in module java.base of loader 'bootstrap')
```

Reply:
> There could also hypothetically be a query that checks for the aggregate count of number of recordings with the well-known name that are currently in the Running state, and this can control whether the Stop Recording action is enabled. Might be overkill. 

Actually very nice idea but right now, the way its set up is not very neat to support this. We will need to go with `regardless of state`.

Reply:
> ### With Metadata/Labels
> 
> ```
> Invalid Syntax : offending token '\"cryostat.io/topology-group\"' at line 1 column 283
> ```
> 
> ### With restart option
> 
> ```
> Exception while fetching data (/environmentNodes[0]/descendantTargets[0]/doStartRecording) : class java.lang.Boolean cannot be cast to class java.lang.String (java.lang.Boolean and java.lang.String are in module java.base of loader 'bootstrap')
> ```

It looks like I probably just gave a bad example of a label key that isn't acceptable, and I guess the "restart" parameter needs to be supplied as a string value, so just wrapping the `true` boolean literal in double quotes (`"true"`) should fix the casting issue I think.

Reply:
Unfortunately, there is another error when wrapping that boolean in string:

```
"Validation error (WrongType@[environmentNodes/descendantTargets/doStartRecording]) : argument 'recording.restart' with value 'StringValue{value='true'}' is not a valid 'Boolean' - Expected AST type 'BooleanValue' but was 'StringValue'."
```

Reply:
> It looks like I probably just gave a bad example of a label key that isn't acceptable.

Seems to work fine when creating recordings. Is there some constraints on the GraphQL side?

![Screenshot from 2023-03-13 19-37-14](https://user-images.githubusercontent.com/68053619/224855417-d6e9415e-f986-4342-bf5f-36388a72df6c.png)

Reply:
What if you also pass the label value in as a query variable? I seem to recall there are some limitations around this on the GraphQL side but I'm not sure if there is a way to escape it here.

For the `restart` thing, it looks like the Mutator code on the backend is just wrongly assuming that it will get a String back from the context object, but GraphQL has already converted iti to a Boolean so the cast fails. The GraphQL schema does specify it should be a Boolean parameter however, so the request validation fails when you wrap it as a String. Should be an easy fix on the backend.

Reply:
Still cant solve this labels but I did some digging and found this PR. Could this be the issue?

https://github.com/cryostatio/cryostat/pull/971#issuecomment-1157956641

Reply:
If it is really, I am thinking we can go with:

```ts
metadata: {
  labels: {
    topology_group: "${group.name.replace(/[\s+-]/g, '_')}"
  }
},
```